src/v3/obfuscate.test.ts

```diff
@@ -0,0 +1,161 @@
+import { obfuscateDocument, obfuscateData } from "./obfuscate";
+import { validateSchema, verifySignature, wrapDocument } from "../";
+import { get } from "lodash";
+import { decodeSalt } from "./wrap";
+
+import { SchemaId } from "../shared/@types/document";
+import { Method, OaProofType, TemplateType, OpenAttestationCredential } from "../../src/__generated__/schemaV3";
+import { toBuffer } from "../shared/utils";
+
+const openAttestationData: OpenAttestationCredential = {
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://www.w3.org/2018/credentials/examples/v1",
+    "https://nebulis.github.io/tmp-jsonld/OpenAttestation.v3.jsonld",
+    "https://nebulis.github.io/tmp-jsonld/CustomContext.jsonld"
+  ],
+  issuanceDate: "2010-01-01T19:23:24Z",
+  name: "document owner name",
+  type: ["VerifiableCredential", "AlumniCredential"],
+  credentialSubject: {
+    id: "did:example:ebfeb1f712ebc6f1c276e12ec21",
+    alumniOf: "Example University"
+  },
+  template: {
+    name: "any",
+    type: TemplateType.EmbeddedRenderer,
+    url: "http://some.example.com"
+  },
+  issuer: "https://example.edu/issuers/14",
+  oaProof: {
+    type: OaProofType.OpenAttestationProofMethod,
+    value: "0x9178F546D3FF57D7A6352bD61B80cCCD46199C2d",
+    method: Method.TokenRegistry
+  }
+};
+
+const testData = {
+  key1: "value1",
+  key2: "value2",
+  ...openAttestationData
+};
+
+describe("privacy", () => {
+  describe("obfuscated", () => {
+    test("removes one field", async () => {
+      const field = "key1";
+      const newDocument = await wrapDocument(testData, { version: SchemaId.v3 });
+      const { data, obfuscatedData } = obfuscateData(newDocument, field);
+      const salt = decodeSalt(newDocument.proof.salts).find(s => s.path === field);
+      const value = get(newDocument, field);
+      expect(obfuscatedData).toEqual([toBuffer({ [field]: `${salt?.value}:${value}` }).toString("hex")]);
+      expect(decodeSalt(data.proof.salts)).not.toHaveProperty(field);
+      expect(data).not.toHaveProperty(field);
+    });
+
+    test("removes multiple fields", async () => {
+      const fields = ["key1", "key2"];
+      const newDocument = await wrapDocument(testData, { version: SchemaId.v3 });
+      const { data, obfuscatedData } = obfuscateData(newDocument, fields);
+      const salts = decodeSalt(newDocument.proof.salts);
+      const salt1 = salts.find(s => s.path === fields[0]);
+      const salt2 = salts.find(s => s.path === fields[1]);
+      const value1 = get(newDocument, fields[0]);
+      const value2 = get(newDocument, fields[1]);
+
+      expect(obfuscatedData).toEqual([
+        toBuffer({ [fields[0]]: `${salt1?.value}:${value1}` }).toString("hex"),
+        toBuffer({ [fields[1]]: `${salt2?.value}:${value2}` }).toString("hex")
+      ]);
+      expect(decodeSalt(data.proof.salts)).not.toHaveProperty(fields);
```

Comment:
according to the doc, this is wrong and perform deep checking (key1.key2) => https://jestjs.io/docs/en/expect#tohavepropertykeypath-value

Reply:
Ohh whoops sorry, changed!